__tests__/nc-news8.test.js

```diff
@@ -12,7 +12,53 @@ describe("PATCH /api/articles/:article_id", () => {
   test("should return 200 status code if sent an object with correct data", () => {
     const newVote = 1;
     const updateVote = { inc_votes: newVote };
-
-    return request(app).patch("/api/articles/4").send(updateVote);
+    return request(app).patch("/api/articles/1").send(updateVote).expect(201);
+  });
+  test("should add votes to an article using article_id", () => {
+    const newVote = 10;
+    const udpateVote = { inc_votes: newVote };
+    return request(app)
+      .get("/api/articles/1")
+      .then(({ body }) => {
+        const { articles } = body;
+        return articles.votes;
+      })
+      .then((currentVotes) => {
+        return request(app)
+          .patch("/api/articles/1")
+          .send(udpateVote)
+          .expect(201)
+          .then(({ body }) => {
+            const { updatedArticle } = body;
+            expect(currentVotes + newVote).toEqual(updatedArticle.votes);
+          });
+      });
+  });
+  test("should subtract votes from an article using article_id", () => {
+    const newVote = -10;
+    const updateVote = { inc_votes: newVote };
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        return articles.votes;
+      })
+      .then((currentVotes) => {
+        return request(app)
+          .patch("/api/articles/1")
+          .send(updateVote)
+          .expect(201)
+          .then(({ body }) => {
+            const { updatedArticle } = body;
+            expect(currentVotes + newVote).toEqual(updatedArticle.votes);
+          });
+      });
+  });
+  test("should return 400 if the article_id is a string", () => {
+    return request(app).get("/api/articles/banana").expect(400);
+  });
+  test("should return 400 if the article_id is an invalid number", () => {
+    return request(app).get("/api/articles/9999").expect(400);
```

Comment:
We should be getting a message with both of these errors too!

Reply:
this gets handled by the errorMiddleware file.

Reply:
Great, but we do need to test for that message :)